### ISSUE TITLE
column ttl does not work

### ISSUE THREAD

**Issue by liwei225** [ClickHouse/ClickHouse#28558]
system: CentOS Linux release 7.9.2009 (Core)

clickhouse version: 21.8.5.7

issue: TTL on column does not work

for example:

![image](https://user-images.githubusercontent.com/22341888/131986803-e328ef04-36e6-4f7a-9148-00e9189169fd.png)

![image](https://user-images.githubusercontent.com/22341888/131986869-fe44e89a-e88d-4a85-beb4-6c877e6f5c58.png)

more than 5 seconds later ,the 'name' column does not clear

![image](https://user-images.githubusercontent.com/22341888/131987005-cc827642-36f1-4f8c-a6e4-898c30cf26e0.png)

optimize does not work too

![image](https://user-images.githubusercontent.com/22341888/131987089-70568ee5-ab26-4664-b977-e49290551343.png)

**Comment by CurtizJ** [comment]
Actually it's not a but, but an unexpected behaviour. When column TTL is processing, the column is dropped by removing whole file, if it's fully empty. However for compact parts it's not done, because we will have to rewrite whole part. So, in your synthetic example with few rows parts parts are created in compact format and data is not dropped. But in real production it's not a problem, because compact format of parts is supposed for little data and can appear only immediately after inserts until part will participate in several merges and it's unlikely, that part will be still in compact format, when TTL will be expired. 

However, looks like this behaviour is fixed in 21.10 (current master) in #27019.

**Comment by CurtizJ** [comment]
The test case:
```sql
DROP TABLE IF EXISTS test.test;
CREATE TABLE test.test (id UInt32, date DateTime, s String DEFAULT 'none' TTL date + INTERVAL 3 SECOND)
ENGINE = MergeTree ORDER BY id;

INSERT INTO test.test VALUES (1, now(), 'test1'), (2, now(), 'test2');

SELECT sleep(3) FORMAT Null;
SELECT sleep(3) FORMAT Null;

SELECT id, s FROM test.test;
```

In 21.8:
```
1       test1
2       test2
```

In master:
```
1       none
2       none
```

The test case:
```sql
DROP TABLE IF EXISTS test.test;
CREATE TABLE test.test (id UInt32, date DateTime, s String DEFAULT 'none' TTL date + INTERVAL 3 SECOND)
ENGINE = MergeTree ORDER BY id
SETTINGS min_bytes_for_wide_part = 0;

INSERT INTO test.test VALUES (1, now(), 'test1'), (2, now(), 'test2');

SELECT sleep(3) FORMAT Null;
SELECT sleep(3) FORMAT Null;

SELECT id, s FROM test.test;
```

Both in 21.8 and master:
```
1       none
2       none
```
